src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/FootnoteServiceTests.cs

```diff
@@ -1205,13 +1916,110 @@ await using (var statisticsDbContext = InMemoryStatisticsDbContext(contextId))
             }
         }
 
+        private static async Task SeedDatabase(string contextId, Release release, IReadOnlyList<Subject>? subjects = null, IReadOnlyList<ReleaseSubject> releaseSubjects = null, IReadOnlyList<Footnote>? footnotes = null)
+        {
+            await using (var contentDbContext = InMemoryApplicationDbContext(contextId))
+            await using (var statisticsDbContext = InMemoryStatisticsDbContext(contextId))
+            {
+                await statisticsDbContext.Release.AddAsync(release);
+
+                if (!subjects.IsNullOrEmpty())
+                {
+                    await statisticsDbContext.Subject.AddRangeAsync(subjects!);
+                }
+
+                if (!releaseSubjects.IsNullOrEmpty())
+                {
+                    await statisticsDbContext.ReleaseSubject.AddRangeAsync(releaseSubjects!);
+                }
+
+                if (!footnotes.IsNullOrEmpty())
+                {
+                    await statisticsDbContext.Footnote.AddRangeAsync(footnotes!);
+                }
+
+                await statisticsDbContext.SaveChangesAsync();
+
+                await contentDbContext.AddAsync(new Content.Model.Release
+                {
+                    Id = release.Id
+                });
+                await contentDbContext.SaveChangesAsync();
+            }
+        }
+
+        private static async Task<Either<ActionResult, Footnote>> CreateFootnoteWithConfiguration(
+            Guid releaseId,
+            string contextId,
+            string content,
+            IReadOnlySet<Guid> filterIds,
+            IReadOnlySet<Guid> filterGroupIds,
+            IReadOnlySet<Guid> filterItemIds,
+            IReadOnlySet<Guid> indicatorIds,
+            IReadOnlySet<Guid> subjectIds)
+        {
+            var dataBlockService = new Mock<IDataBlockService>();
+            dataBlockService
+                .Setup(dbs => dbs.InvalidateCachedDataBlocks(releaseId))
+                .Verifiable();
+
+            await using (var contentDbContext = InMemoryApplicationDbContext(contextId))
+            await using (var statisticsDbContext = InMemoryStatisticsDbContext(contextId))
+            {
+                FootnoteService footnoteService = SetupFootnoteService(contentDbContext: contentDbContext, statisticsDbContext: statisticsDbContext, dataBlockService: dataBlockService.Object);
```

Comment:
I think here I'd technically still say that a concrete type would be best, because `SetupFootnoteService` might not necessarily return a `FootnoteService` even though it's named like it should. I'm happy to follow any coding standards we have though if we have already agreed that we do this if the method name is obvious?

Reply:
I'd usually only put `var` if the Right-hand-side is **explicit** about the type

Reply:
going to discuss in backend guild meeting

Reply:
Discussed further with @jack-hive the arguments for and against using `var`. The arguments for declaring the type name in terms of readability are pretty convincing. It's obvious what the type of the variable is.

The pro's for `var` are also pretty string though:

- `var` is shorter - I think historically we've preferred brevity over verbosity on the project.
- `var` encourages you more to think of a descriptive variable name.
- `var` requires less code changes when the type of a method call changes.

It's also the default in Rider, see [Enforce preferences of using 'var' keyword](https://www.jetbrains.com/help/rider/Using_var_Keyword_in_Declarations.html#enforcing-preferences-of-using-var-keyword), but it can easily be changed. I think this will be another reason why we've preferred using `var`.

We said that we can't change this without taking it to the BE style guide meeting.

Jack will leave the code as it is for now and update it to use `var` later depending on the outcome of the meeting.